### ISSUE TITLE
Typecheck error

### ISSUE THREAD

**Issue by hvoss49** [zauguin/luapstricks#73]
! luapstricks: "typecheck" error occured while executing PS code from "./mein.tex:10".
<inserted text> BAD
   
l.19 \end{document}

up-to-date luapstricks, The output is correct.

```
\RequirePackage{pdfmanagement-testphase}
\DeclareDocumentMetadata{}
\documentclass[11pt]{report}
\usepackage{pst-plot,pst-intersect,mathtools}
\begin{document}

\begin{pspicture}(-2,-2)(8,8)
\psaxes[labelFontSize=\scriptstyle,ticksize=0 4pt]{->}(0,0)(-2,-2)(8,8)[$M$,-90][$Y$,0]
\psset{linewidth=1.5pt,algebraic}
\pssavepath[linecolor=green]{A}{\psplot{-0.5}{8}{4*(1-1.2^(-3*x+1))}}
\psline[linecolor=red](-2,4)(8,4) \uput[90](7,4){$Y=\dfrac{A}{\alpha+d}$}
%\pssavepath[linecolor=blue]{B}{\psplot{-0.5}{8}{2^(-x/2+3)-2}}
%\pssavepath[linestyle=none]{C}{\psplot{-0.5}{8}{0}}
%\psintersect[name=D, showpoints]{A}{B}\uput[0](D1){$M_3^*,Y^*$}
%psintersect[name=E, showpoints]{A}{C}\uput[-45](E1){$M_c$}
%\psdot(4,0)\uput[225](4,0){$M_c^*$}
\end{pspicture}

\end{document} 
```

**Comment by zauguin** [comment]
The problem is that `pst-intersect` tries to put some elements onto the stack in one special and then retrieve them in the next. We clear all elements left over from one special in order to be compatible with the default behavior of `@endspecial`. Now `pst-intersect` changes the counters used by `@endspecial` and therefore avoids `@endspecial` from breaking this, while our counter is not affected.

I'll try to change `luapstricks` to use `@beginspecial`/`@endspecial`'s `ocount` counter too to support this, but that might be a bit complicated since it triggers some nesting related issues.

**Comment by zauguin** [comment]
Should be fixed in 72e8d8208b91fcc2a7443e7514bd9c106d415f4f.

**Comment by hvoss49** [comment]
works! Thanks
